src/hooks/usePushNotifications.tsx

```diff
@@ -0,0 +1,108 @@
+import React, {
+  Dispatch,
+  SetStateAction,
+  createContext,
+  useEffect,
+  useState,
+} from 'react';
+import {
+  getInitialNotification,
+  onNotificationOpened,
+  onNotificationReceived,
+} from '../../src/common';
+import { Platform } from 'react-native';
+import { Notifications } from 'react-native-notifications';
+
+type PushNotificationsEventType = 'notificationReceived' | 'notificationOpened';
+
+export type PushNotificationsEvent = {
+  type: PushNotificationsEventType;
+  notification: Notification;
+};
+
+interface PushNotificationsStateType {
+  events: PushNotificationsEvent[];
+  setEvents: Dispatch<SetStateAction<PushNotificationsEvent[]>>;
+}
+
+export const PushNotificationsContext =
+  createContext<PushNotificationsStateType>({
+    events: [],
+    setEvents: () => [],
```

Comment:
Can you explain how you're planning on using `events` and `setEvents` once they're exposed?

Reply:
these are just hooks for the push notification event queue - so by having these from a provider a user can have a button to clear/reset their notification queue for example

Reply:
I'm confused by this provider's use case. I'd imagine we'd use it to register the device for PN, which is not the case currently. We can also use it to expose events such as notifications received or notifications opened.
However, I don't see how `setEvents` would be used. Can you elaborate on why would a developer need to be able to set the events list internally in the provider? For example, a developer can do something like:
`setEvents([id: '1', title: 'I never received this notification, but I'm setting it in the provider'])`.